api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseDaoImpl.java

```diff
@@ -424,6 +431,62 @@ protected boolean containsAlias(Iterator<CriteriaImpl.Subcriteria> subcriteriaIt
 		return Optional.empty();
 	}
 	
+	protected Optional<Criterion> handleQuantity(String propertyName, QuantityParam quantityParam) {
+		if (quantityParam == null) {
+			return Optional.empty();
+		}
+		
+		BigDecimal value = quantityParam.getValue();
+		BigDecimal approxRange = APPROX_RANGE.multiply(value);
```

Comment:
This should be pushed down to where `dotIdx == -1` since that's the only place we need this value